src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java

```diff
@@ -100,6 +120,35 @@ public void patchCompanyNameStatus(String companyNumber, String companyName,
         throw new UnsupportedOperationException("Method not implemented");
     }
 
+    public void patchNewAppointmentCompanyNameStatus(String companyNumber, String appointmentId, String companyName,
+            String companyStatus, String contextId) throws BadRequestException, NotFoundException, ServiceUnavailableException {
+        if (isBlank(companyName) || isBlank(companyStatus)) {
+            throw new BadRequestException("Request missing mandatory fields: company name and/or company status");
```

Comment:
Could we log the appointmentId so that each request with the same error has a unique log message?

Reply:
Addressed in latest commit